textsplitter/split_documents.go

```diff
@@ -1,65 +1,36 @@
 package textsplitter
 
 import (
-	"errors"
 	"log"
 	"strings"
 
 	"github.com/tmc/langchaingo/schema"
 )
 
-// ErrMismatchMetadatasAndText is returned when the number of texts and metadatas
-// given to CreateDocuments does not match. The function will not error if the
-// length of the metadatas slice is zero.
-var ErrMismatchMetadatasAndText = errors.New("number of texts and metadatas does not match")
```

Comment:
Why are these changes in this PR? Are they essential to support custom ID? Can they be in a separate PR?

Reply:
These are needed as when you split documents using a text splitter, we need to preserve the custom ID.

That lead me to refactoring the `SplitDocuments` function to make it simpler. After this refactor, this error was no longer possible, hence it was removed.

Reply:
I suggest moving this to a separate PR, which can come either before or after the custom ID PR - up to you. Keeping one PR per topic makes it much easier to do a thorough review